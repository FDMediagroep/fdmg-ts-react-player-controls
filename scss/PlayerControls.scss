.player-controls {
  display: flex;
  align-items: baseline;
  justify-content: space-between;
  max-width: 130px;
  margin: 0 auto;
  a {
    width: 30px;
    height: 30px;
    display: block;
    background-repeat: no-repeat;
    background-size: contain;
    &:hover {
      cursor: pointer;
    }
    &.previous-button, &.next-button {
      background-size: 22px 22px;
      background-position-y: 4px;
      background-position-x: 4px;
    }
    &.play-button, &.pause-button {
      background-size: 30px 30px;
    }
    &.previous-button {
      background-image: url("../img/playercontrols/previous-button.svg");
    }
    &.next-button {
      background-image: url("../img/playercontrols/next-button.svg");
    }
    &.play-button {
      background-image: url("../img/playercontrols/play-button.svg");
    }
    &.pause-button {
      background-image: url("../img/playercontrols/pause-button.svg");
    }
  }
}

.IE10 {
  .player-controls {
    .next-button, .previous-button {
      img {
        padding: 5px;
        max-width: 20px;
      }
    }
    .playpause-button {
      img {
        max-width: 30px;
      }
      margin: 0 10px;
    }
    div {
      display: block;
      text-align: center;
    }
  }
  .counters {
    display: block;
    height: 12px;
    span {
      &:first-child {
        float: left;
      }
      &:last-child {
        float: right;
      }
    }
  }
}