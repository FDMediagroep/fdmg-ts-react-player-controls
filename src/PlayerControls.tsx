import * as React from 'react';
import ImageButton from "fdmg-ts-react-image-button";

export interface PlayerControlsProps {
    className?: string;
    showPreviousButton: boolean;
    showPlayPauseButton: boolean;
    showNextButton: boolean;
    previousButtonImage?: string;
    playPauseButtonImage?: string;
    nextButtonImage?: string;
    onPrevious?: string;
    onPlayPause?: string;
    onNext?: string;
}

/**
 * PlayerControls composition - Renders the player's
 * previous, play/pause and next button. 
 */
export default class PlayerControls extends React.Component<PlayerControlsProps, any> {
    public state: any;
    public props: any;

    constructor(props: PlayerControlsProps) {
        super(props);
        this.props = props;
        console.info('Player controls composition', this.state, this.props);        
    }

    componentDidMount() {
        console.info('Player controls mounted', this.props);
    }
        
    componentWillUnmount() {
        console.info('Player controls will unmount');
    }

    componentWillReceiveProps(nextProps) {
        console.info('Player controls next props', nextProps);
    }

    render() {
        return (
            <div className={this.props.className}>
                {this.props.showPreviousButton?
                    <ImageButton
                        src={this.props.previousButtonImage}
                        onClick={this.props.onPrevious}
                        className="previous-button"
                        alt="previous button"
                    />
                    :null
                }

                {this.props.showPlayPauseButton?
                    <ImageButton
                        src={this.props.playPauseButtonImage}
                        onClick={this.props.onPlayPause}
                        className="playpause-button"
                        alt="play/pause button"
                    />
                    :null
                }

                {this.props.showNextButton?
                    <ImageButton
                        src={this.props.nextButtonImage}
                        onClick={this.props.onNext}
                        className="next-button"
                        alt="next button"
                    />
                    :null
                }
            </div>
        );
    }
}
 