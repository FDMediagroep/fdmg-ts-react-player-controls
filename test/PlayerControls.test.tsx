import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import * as React from 'react';
import PlayerControls from "../src/PlayerControls";

console.info = () => {};

describe('PlayerControls', () => {
    const emptyFn = () => {};

    beforeAll(() => {
        Enzyme.configure({ adapter: new Adapter() });
    });

    test('PlayerControls renders correctly', () => {
        const component = shallow(
            <PlayerControls
                showPreviousButton={true}
                showPlayPauseButton={true}
                showNextButton={true}
                onPrevious={emptyFn}
                onPlayPause={emptyFn}
                onNext={emptyFn}
            />
        );
        expect(toJson(component)).toMatchSnapshot();

        component.setProps({playPauseButtonImage: 'img/playercontrols/play-button.tsx'});
        expect(toJson(component)).toMatchSnapshot();
    });

    test('PlayerControls buttons clicked', () => {
        const spyPrev = jest.fn();
        const spyPlay = jest.fn();
        const spyNext = jest.fn();
        const component = mount(
            <PlayerControls
                showPreviousButton={true}
                showPlayPauseButton={true}
                showNextButton={true}
                onPrevious={spyPrev}
                onPlayPause={spyPlay}
                onNext={spyNext}
            />
        );

        component.find('span.next-button').simulate('click');
        expect(spyNext).toHaveBeenCalled();
        component.find('span.playpause-button').simulate('click');
        expect(spyPlay).toHaveBeenCalled();
        component.find('span.previous-button').simulate('click');
        expect(spyPrev).toHaveBeenCalled();
    });

    test('PlayerControls renders previous, next or play/pause button correctly', () => {
        let component = mount(
            <PlayerControls
                showPreviousButton={true}
                showPlayPauseButton={false}
                showNextButton={false}
                onPrevious={emptyFn}
                onPlayPause={emptyFn}
                onNext={emptyFn}
            />
        );
        expect(toJson(component)).toMatchSnapshot();

        component = mount(
            <PlayerControls
                showPreviousButton={false}
                showPlayPauseButton={true}
                showNextButton={false}
                onPrevious={emptyFn}
                onPlayPause={emptyFn}
                onNext={emptyFn}
            />
        );
        expect(toJson(component)).toMatchSnapshot();

        component = mount(
            <PlayerControls
                showPreviousButton={false}
                showPlayPauseButton={false}
                showNextButton={true}
                onPrevious={emptyFn}
                onPlayPause={emptyFn}
                onNext={emptyFn}
            />
        );
        expect(toJson(component)).toMatchSnapshot();

    });
});
