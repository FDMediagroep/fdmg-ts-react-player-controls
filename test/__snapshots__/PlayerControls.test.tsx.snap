// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerControls renders correctly 1`] = `
<div>
  <ImageButton
    alt="previous button"
    className="previous-button"
    onClick={[Function]}
  />
  <ImageButton
    alt="play/pause button"
    className="playpause-button"
    onClick={[Function]}
  />
  <ImageButton
    alt="next button"
    className="next-button"
    onClick={[Function]}
  />
</div>
`;

exports[`PlayerControls renders correctly 2`] = `
<div>
  <ImageButton
    alt="previous button"
    className="previous-button"
    onClick={[Function]}
  />
  <ImageButton
    alt="play/pause button"
    className="playpause-button"
    onClick={[Function]}
    src="img/playercontrols/play-button.tsx"
  />
  <ImageButton
    alt="next button"
    className="next-button"
    onClick={[Function]}
  />
</div>
`;

exports[`PlayerControls renders previous, next or play/pause button correctly 1`] = `
<div>
  <ImageButton
    alt="previous button"
    className="previous-button"
    onClick={[Function]}
  />
</div>
`;

exports[`PlayerControls renders previous, next or play/pause button correctly 2`] = `
<div>
  <ImageButton
    alt="play/pause button"
    className="playpause-button"
    onClick={[Function]}
  />
</div>
`;

exports[`PlayerControls renders previous, next or play/pause button correctly 3`] = `
<div>
  <ImageButton
    alt="next button"
    className="next-button"
    onClick={[Function]}
  />
</div>
`;
